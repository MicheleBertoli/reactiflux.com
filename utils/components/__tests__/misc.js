import React from 'react';
import renderer from 'react-test-renderer';
import { matcher, serializer } from 'jest-styled-components'
import {
  ReadMore,
  Copyright,
  Credits,
  PublishDate,
  Transcript,
  Logo,
} from '../misc';

expect.extend(matcher)
expect.addSnapshotSerializer(serializer)

test('<ReadMore />', () => {
  const tree = renderer.create(<ReadMore />).toJSON();

  expect(tree).toMatchStyledComponentsSnapshot();
});

test('<Copyright />', () => {
  const tree = renderer.create(<Copyright />).toJSON();

  expect(tree).toMatchStyledComponentsSnapshot();
});

test('<Credits />', () => {
  const tree = renderer.create(<Credits />).toJSON();

  expect(tree).toMatchStyledComponentsSnapshot();
});

test('<PublishDate />', () => {
  const tree = renderer.create(<PublishDate />).toJSON();

  expect(tree).toMatchStyledComponentsSnapshot();
});

test('<Transcript />', () => {
  const props = {
    article: {
      date: 'date',
      body: 'body',
    },
  };
  const tree = renderer.create(<Transcript {...props} />).toJSON();

  expect(tree).toMatchStyledComponentsSnapshot();
});

test('<Logo />', () => {
  const tree = renderer.create(<Logo />).toJSON();

  expect(tree).toMatchStyledComponentsSnapshot();
});
