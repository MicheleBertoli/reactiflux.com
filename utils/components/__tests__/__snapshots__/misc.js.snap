// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Copyright /> 1`] = `
.fvZRKP {
  color: #52244f;
  line-height: 19px;
  margin-bottom: 6px;
}

<p
  className="fvZRKP"
/>
`;

exports[`<Credits /> 1`] = `
.dPQdYb {
  font-size: 12px;
  line-height: 12px;
  margin-bottom: 6px;
  color: #52244f;
  font-family: 'Space Mono', monospace;
}

<p
  className="dPQdYb"
/>
`;

exports[`<Logo /> 1`] = `
.gmVDZZ {
  display: inline-block;
  width: 59px;
  height: 52px;
  background: url();
  text-indent: -9999px;
  z-index: 3;
}

<a
  className="gmVDZZ"
  onClick={[Function]}
  style={Object {}}
/>
`;

exports[`<PublishDate /> 1`] = `
.gKgEHh {
  font-size: 0.8rem;
  font-style: italic;
}

<time
  className="gKgEHh"
/>
`;

exports[`<ReadMore /> 1`] = `
.UPetn {
  display: none;
  width: 100%;
  margin-bottom: 20px;
  color: #dd1d64;
  justify-content: space-between;
  -webkit-box-pack: justify;
  -ms-flex-pack: justify;
  -webkit-justify-content: space-between;
}

@media (max-width: 768px) {
  .UPetn {
    display: -webkit-box;
    display: -moz-box;
    display: -ms-inline-flexbox;
    display: -webkit-inline-flex;
    display: inline-flex;
  }
}

<div
  className="UPetn"
/>
`;

exports[`<Transcript /> 1`] = `
.iwDvKU {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-inline-flexbox;
  display: -webkit-inline-flex;
  display: inline-flex;
  flex-flow: column;
  -ms-flex-flow: column;
  -webkit-flex-flow: column;
  flex-basis: 70%;
  -ms-preferred-size: 70%;
  -webkit-flex-basis: 70%;
  max-width: 1192px;
}

.gKgEHh {
  font-size: 0.8rem;
  font-style: italic;
}

@media (max-width: 768px) {
  .iwDvKU {
    flex-basis: auto;
    -ms-preferred-size: auto;
    -webkit-flex-basis: auto;
  }
}

<div
  className="markdown iwDvKU"
>
  <time
    className="gKgEHh"
  >
    Transcript from 
    date
  </time>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "body",
      }
    }
  />
</div>
`;
